### ISSUE TITLE
PETSC failure on some RHEL machines

### ISSUE THREAD

**Issue by jacobmerson** [SCOREC/rhel7-dev#3]
Petsc VecAYPX is giving a segmentation fault on Jenga. The same code seems to work fine on the RHEL7 builds we have for Professor Picu's lab.

Note: I am using `petsc/3.9.3-int32-real-c-ceyty2w`

**Comment by jacobmerson** [comment]
@wrtobin mentioned that he was going to run the petsc tests as part of the spack build. It would be useful to know if those tests ended up passing.

Also as we discussed, currently petsc is compiling with dyamic libs. We may want to only have static libs for compatibility of our build process with BGQ.

**Comment by wrtobin** [comment]
The PETSc spack build currently does not complete when the tests are configured to run for some reason, possibly a bug of some sort. I may be able to look into it more later this week after the fusion meeting Thurs morning.

**Comment by jacobmerson** [comment]
It seems like the superlu config may be part of the issue.

**Comment by wrtobin** [comment]
Yeah this is still an issue. After looking more explicitly at this and making the 'make test' call myself on the spack-created configure/build directory for PETSc, all the tests are failing.

**Comment by wrtobin** [comment]
This is on Jenga since that is my office machine. I haven't looked on any other machines. Can you point me to a machine in Picu's lab that you've noted is working? @jacobmerson

**Comment by wrtobin** [comment]
Tagging @cwsmith here cause I also just emailed him about this issue.

**Comment by jacobmerson** [comment]
I had petsc working on backgammon and petsc, however as of this week I was having trouble compiling on backgammon due to limits.h missing...

**Comment by wrtobin** [comment]
Oh you compiled your own PETSc? Have you successfully used the spack-installed PETSc to link and execute any application binaries without encountering a memory violation error?

**Comment by jacobmerson** [comment]
I used to use my own version, but the spack version worked on those machines.

**Comment by wrtobin** [comment]
Okay I'll try executing the PETSc test suite on one of those

**Comment by wrtobin** [comment]
The spack testing still fails with the same issue, but explicitly making the test call succeeds on backgammon.

**Comment by wrtobin** [comment]
Soooooooooooo I'm confused.

**Comment by wrtobin** [comment]
Calling 'make test' on backgammon succeeds for all tests. 

For an identical spack config on jenga 'make test' fails.

**Comment by wrtobin** [comment]
I've submitted a support ticket trying to describe this issue in detail to dotcio (aka Bob), because I'm currently at a loss for how to proceed and until this works I'm stuck on development of some things for m3dc1.

**Comment by wrtobin** [comment]
The default spack version of PETSc is now 3.10.2, and the installation of this version (on Jenga) appears to pass PETSc's testing suite during build. 

@cwsmith Do we have a substantial number of people using the 3.9.3 version that you know of, and do you think switching our install to the 3.10.2 version and possibly removing the (seemingly unstable) 3.9.3 version might be the way to go?

**Comment by cwsmith** [comment]
@wrtobin AFAIK, the only petsc users were XGCm developers; feel free to replace the existing installs.

**Comment by cwsmith** [comment]
As I understand, part of the issue was incompatibility with the installed version of (par)metis. All the gcc 7.3.0 and mpich 3.1 spack packages were removed (for a different reason) and PETSc 3.11 was reinstalled prior to the other gcc 7.3.0 and mpich 3.3 spack packages.

Would you please rerun any of the problematic cases/tests?

**Comment by jacobmerson** [comment]
@wrtobin can you rerun the petsc test suite? I will rerun the original offending Biotissue code, but it might take me a couple days to get to it.
